source/val/validation_state.h

```diff
@@ -235,38 +247,66 @@ class ValidationState_t {
   }
 
   /// Registers execution mode for the given entry point.
-  void RegisterExecutionModeForEntryPoint(uint32_t entry_point,
-                                          spv::ExecutionMode execution_mode) {
-    entry_point_to_execution_modes_[entry_point].insert(execution_mode);
+  void RegisterExecutionModeForEntryPoint(const Instruction* inst) {
+    const uint32_t entry_point = inst->word(1);
+    const spv::ExecutionMode mode = spv::ExecutionMode(inst->word(2));
+    entry_point_data_[entry_point].execution_modes.insert(mode);
+
+    // Save for now since the IDs might have not been parsed yet
```

Comment:
> Still thinking about the change to save execution mode instructions. I'm not so sure about that.

I guess is the issue there is some subtle reason we shouldn't be tracking the `Instructions` ?

Reply:
ok, I will need to cache this PR back into my brain, but this makes sense to me, will try and adjust it this week